code/game/objects/pool/pool.dm

```diff
@@ -0,0 +1,308 @@
+*/
+
+GLOBAL_LIST_EMPTY(species_swimming_components)
+
+/obj/effect/overlay/poolwater
+	name = "Pool water"
+	icon = 'icons/obj/pool.dmi'
+	icon_state = "water"
+	anchored = TRUE
+	layer = ABOVE_ALL_MOB_LAYER
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+
+/turf/open/indestructible/sound/pool
+	name = "Swimming pool"
+	desc = "A fun place where you go to swim! <b>Drag and drop yourself onto it to climb in...</b>"
+	icon = 'icons/obj/pool.dmi'
+	icon_state = "pool"
+	sound = 'sound/effects/splash.ogg'
+	var/id = null //Set me if you don't want the pool and the pump to be in the same area, or you have multiple pools per area.
+	var/obj/effect/water_overlay = null
+
+/turf/open/indestructible/sound/pool/end
+	icon_state = "poolwall"
+
+/turf/open/indestructible/sound/pool/Initialize(mapload)
+	. = ..()
+	water_overlay = new /obj/effect/overlay/poolwater(get_turf(src))
+	if(!LAZYLEN(GLOB.species_swimming_components))
+		//Setup the swimming species
+		GLOB.species_swimming_components[/datum/species/squid] = /datum/component/swimming/squid
+		GLOB.species_swimming_components[/datum/species/ethereal] = /datum/component/swimming/ethereal
+		GLOB.species_swimming_components[/datum/species/jelly] = /datum/component/swimming/disolve
+		for(var/golem_type in typesof(/datum/species/golem))
+			GLOB.species_swimming_components[golem_type] = /datum/component/swimming/golem
+
+/turf/open/indestructible/sound/pool/proc/set_colour(colour)
+	water_overlay.color = colour
+
+/turf/open/indestructible/sound/pool/end/ChangeTurf(path, list/new_baseturfs, flags)
+	if(water_overlay)
+		qdel(water_overlay)
+	. = ..()
+
+/turf/open/CanPass(atom/movable/mover, turf/target)
+	var/datum/component/swimming/S = mover.GetComponent(/datum/component/swimming) //If you're swimming around, you don't really want to stop swimming just like that do you?
+	if(S)
+		return FALSE //If you're swimming, you can't swim into a regular turf, y'dig?
+	. = ..()
+
+/turf/open/indestructible/sound/pool/CanPass(atom/movable/mover, turf/target)
+	var/datum/component/swimming/S = mover.GetComponent(/datum/component/swimming) //You can't get in the pool unless you're swimming.
+	return (isliving(mover)) ? S : ..() //So you can do stuff like throw beach balls around the pool!
+
+/turf/open/indestructible/sound/pool/Entered(atom/movable/AM)
+	. = ..()
+	SEND_SIGNAL(AM, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_WEAK)
+	if(isliving(AM))
+		var/datum/component/swimming/S = AM.GetComponent(/datum/component/swimming) //You can't get in the pool unless you're swimming.
+		if(!S)
+			var/mob/living/carbon/C = AM
+			var/component_type = /datum/component/swimming
+			if(istype(C))
+				if(C.dna.species.type in GLOB.species_swimming_components)
+					component_type = GLOB.species_swimming_components[C.dna.species.type]
+				else if(iscatperson(C))	//Why is this not it's own species, that is really annoying
+					component_type = /datum/component/swimming/felinid
```

Comment:
Cat person has their own species file, surely it would just be /datum/species/felinid